### PR TITLE
[main] Fixes for agentEnvVars, updates for proxy tests

### PR DESCRIPTION
**Pull request by MSpencer87**
Fixes for custom cluster reg strings with env vars

Added helper for setting corral bastion addresses

### CODE REVIEW

**Comment by slickwarren** [comment]
can you link the rancher/rancher PR here for reference to the updated structs?

File: extensions/clusters/clusters.go [hunk 1]
```diff
@@ -305,6 +306,10 @@ func NewRKE1ClusterConfig(clusterName string, client *rancher.Client, clustersCo
 		newConfig.DefaultPodSecurityAdmissionConfigurationTemplateName = clustersConfig.PSACT
 	}
 
+	if clustersConfig.AgentEnvVars != nil {
+		newConfig.AgentEnvVars = clustersConfig.AgentEnvVarsRKE1
+	}
+
```
Comment (slickwarren) [inline extensions/clusters/clusters.go:312]: this is already set on line 270, do we need the if != nil logic here?

File: extensions/clusters/clusters.go [hunk 2]
```diff
@@ -450,19 +455,26 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *Clus
 		MachinePools:         machinePools,
 	}
 
+	agentEnvVars := []rkev1.EnvVar{}
```
Comment (slickwarren) [inline extensions/clusters/clusters.go:458]: can't this be omitted, or does setting an initial object match more closely to what rancher does to create a cluster? (thinking of the case where no env vars are set, this was nil before this PR)

**Review by slickwarren — COMMENTED** [review]
mostly looks solid, just some comments on possibly redundant initializations

**Review by markusewalker — APPROVED** [review]
LGTM

File: extensions/clusters/clusters.go (hunk 1 above)
Comment (igomez06) [reply extensions/clusters/clusters.go:314]: it could be nil right Michael?

File: extensions/clusters/clusters.go (hunk 2 above)
Comment (igomez06) [reply extensions/clusters/clusters.go:460]: Good question.

**Review by slickwarren — APPROVED** [review]

**Pull request merged**